### PR TITLE
Disable compaction during restore

### LINKED ISSUE #3953: Disable compaction during restore
As observed in benchmarks, restore (mainly l&s) slows down noticeably when the compaction is running.
The compactions should be disabled during restore and re-enabled afterwards (similar to `tombstone_gc`).
image
image

### PR DESCRIPTION
**Pull request by Michal-Leszczynski**
Fixes #3953

### CODE REVIEW

**Comment by Michal-Leszczynski** [comment]
@karol-kokoszka this PR is ready for review!

File: pkg/managerclient/restore.go [hunk 1]
```diff
@@ -6,21 +6,25 @@ package managerclient
 const (
 	RestoreStageInit          = "INIT"
 	RestoreStageDropViews     = "DROP_VIEWS"
+	StageDisableCompaction    = "DISABLE_COMPACTION"
```
Comment (karol-kokoszka) [inline pkg/managerclient/restore.go:9]: Please follow the same pattern by either removing `Restore` prefix from all the stages, or adding prefix `Restore` to the new stages.

File: pkg/service/restore/restore_integration_test.go [hunk 2]
```diff
@@ -334,3 +338,103 @@ func TestRestoreTablesVnodeToTabletsIntegration(t *testing.T) {
...
+				close(reachedDataStageChan)
+			}
+			<-waitCopyPaths
+		}
+		return nil, nil
+	}))
+
+	Print("Run restore")
+	wg := sync.WaitGroup{}
+	wg.Add(1)
```
Comment (karol-kokoszka) [inline pkg/service/restore/restore_integration_test.go:393]: Please use one approach for syncing the code.
Either sync.WaitGroup{} or synchronization by closing the channel.

For single goroutine, channel is ok.

**Review by karol-kokoszka — COMMENTED** [review]
Two nits.

**Comment by Michal-Leszczynski** [comment]
@karol-kokoszka I'm wondering if we should before merging this PR.
I don't know if we have the evidence that this is the right thing to do for any type of restore.
Perhaps it could be controlled by a new flag (e.g. `--disable-compaction`)?

**Comment by Michal-Leszczynski** [comment]
Closed in favor of #4060.

**Pull request closed without merging**